domain/rates/rates_service.go

```diff
@@ -0,0 +1,88 @@
+package rates
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/bitcoin-sv/spv-wallet-web-backend/config"
+	"github.com/rs/zerolog"
+	"github.com/spf13/viper"
+)
+
+// RatesService is a service for fetching and caching BSV exchange rates.
+type RatesService struct {
+	exchangeRate *ExchangeRate
```

Comment:
Maybe we can just hold the pointer to `float64` instead of using additional struct `ExchangeRate`

Reply:
I cannot unmarshal request data into *float64 it requires struct

2024-05-10 15:04:58.100 ERR error during unmarshalling response body: json: cannot unmarshal object into Go value of type float64 application=spv-wallet-web-backend ecs.version=1.6.0

Reply:
I suppose you've created the `ExchangeRate` struct based on model what is returned from `whatsOnChain`... and it's ok 👍 it's even necessary to properly parse the json. 

But, I meant something slightly different - after the process of unmarshalling, we can hold the rate itself (without struct wrapping) as `*float64`.
- Additionally, the function `GetExchangeRate` could return (float64, error) - that would simplify usage of the whole functionality.
- Moreover, what if we change the rate service provider with different model returned? My suggestion would separate the external usage of the service from its internal implementation/models.

Reply:
Ok, now I got it, good point. Replaced ExchangeRate struct with plain float64 everywhere, struct is now used only for fetching data